### PR TITLE
unison: update version 2.48.4->2.51.2

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS 10.13
Xcode 9.4

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: net/unison/Portfile [hunk 1]
```diff
@@ -16,39 +15,41 @@ long_description    Unison is a file-synchronization tool for Unix and \
...
+supported_archs     i386 x86_64
+universal_variant   no
 
-patchfiles-append   patch-Makefile.diff
-platform darwin 8 {
-    patchfiles-append patch-Makefile.OCaml-MINOSXVERSION.diff
-}
-
-worksrcdir          src
+depends_build       bin:ocamlopt:ocaml
```
Comment (pmetzger) [inline net/unison/Portfile]: I would just depend on the ocaml package. It isn't like we have incompatible packages around.

**Comment by pmetzger** [comment]
Other than the above, LGTM, but I'll confess I didn't read everything very closely.

File: net/unison/Portfile (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply net/unison/Portfile]: You make a good point.
I was jus following what came before (``depends_build  bin:camlp4:ocaml``).
I have made the requested change.

File: net/unison/Portfile [hunk 2]
```diff
@@ -16,39 +15,41 @@ long_description    Unison is a file-synchronization tool for Unix and \
...
-    }
 }
 
-variant x11 conflicts aqua {
-    patchfiles-append       patch-Makefile.OCaml.diff
+build.env-append    CC=${configure.cc}
+destroot.env-append PREFIX=${prefix}
+
+build.args          UISTYLE=text THREADS=true
+destroot.args       UISTYLE=text THREADS=true
```
Comment (pmetzger) [inline net/unison/Portfile]: I note that you're not doing `NATIVE` any more. Was that deprecated? (I don't know Unison but I imagine that means generate a native executable and not OCamlrun bytecode. I would have to check to be sure.)

Comment (MarcusCalhoun-Lopez) [reply net/unison/Portfile]: I believe that ``NATIVE`` is ``true`` by default (see [here](https://github.com/bcpierce00/unison/blob/e1e047b038aed24f7d084e4dbb5b6a2c278e76c2/src/Makefile#L14)).

Comment (pmetzger) [reply net/unison/Portfile]: Gotcha.

**Comment by pmetzger** [comment]
Merged!

**Pull request merged**